tracing: Allow disabling idle traces

ksychla opened the pull request:
This PR adds the possibility of turning off/on the idle traces the same way you can turn off/on other traces. The idle traces can be a bit too verbose, which can clog the tracing ring buffer, leading to some traces not showing up.

ksychla committed:
tracing: Allow disabling idle traces

This change allows for enabling/disabling the idle traces by setting the
CONFIG_TRACING_IDLE config.

Signed-off-by: Krzysztof Sychla <ksychla@antmicro.com>

pdgendt commented on subsys/tracing/Kconfig:
```diff
@@ -356,6 +356,12 @@ config TRACING_GPIO
 
+config TRACING_IDLE
+	bool "Tracing Idle"
+	default n
```
Omit `default n` (can be done too for `TRACING_GPIO`)

```suggestion
```

ksychla commented:
The CI was failing at the cpu_load test, because the cpu_load functionality uses the idle tracing hooks, and when they are disabled, cpu_load doesn't work properly. I added a commit that separates the two, and now the cpu_load hooks are being called directly from the idle functions

nashif commented:
> The CI was failing at the cpu_load test, because the cpu_load functionality uses the idle tracing hooks, and when they are disabled, cpu_load doesn't work properly. I added a commit that separates the two, and now the cpu_load hooks are being called directly from the idle functions

please see https://github.com/zephyrproject-rtos/zephyr/pull/71667#issuecomment-2630676962

ksychla commented:
I removed my changes to the cpu_load subsystem. I added ifdef guards to the tracing format idle functions to mute the idle traces when the config is disabled. So we can still use the cpu_load functionality simultaneously being able to mute the idle traces.
